eth/eventservices/jobservice.go

```diff
@@ -80,6 +80,11 @@ func (s *JobService) Start(ctx context.Context) error {
 		return err
 	}
 
+	s.eventMonitor.SubscribeNewBlock(context.Background(), "BlockWatcher", make(chan *types.Header), func(h *types.Header) (bool, error) {
```

Comment:
Just a comment - I think we should re-architect how we are subscribing to events.  There is currently a lot of duplicated subscriptions.  Every time we call `SubscribeNewBlock`, we send a new subscription request to the remote provider (could explain why our sub connection is unstable).  

Instead, we should be sending a sub request per event type, and having multiple local listeners on the same event.